torchreid/integration/sc/model_wrappers/classification.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (C) 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+import numpy as np
+
+from openvino.model_zoo.model_api.models.image_model import ImageModel
+from openvino.model_zoo.model_api.models.types import NumericalValue, ListValue, StringValue, BooleanValue
+
+
+class Classification(ImageModel):
+    __model__ = 'classification'
+
+    def __init__(self, model_adapter, configuration=None, preload=False):
+        super().__init__(model_adapter, configuration, preload)
+        self._check_io_number(1, 1)
+        if self.path_to_labels:
+            self.labels = self._load_labels(self.path_to_labels)
+
+        self.output_blob_name = self._get_outputs()
+
+    @classmethod
+    def parameters(cls):
+        parameters = super().parameters()
+        parameters.update({
+            'multilabel': BooleanValue(default_value=False),
+            'topk': NumericalValue(value_type=int, default_value=1),
+            'labels': ListValue(description="List of class labels"),
+            'path_to_labels': StringValue(description="Path to file with labels. Overrides the labels, if they sets via 'labels' parameter")
+        })
+
+        return parameters
+
+    @staticmethod
+    def _load_labels(labels_file):
```

Comment:
This plain text format doesn't cover the hierarchical labels case. I'm afraid that we can't support it without ote_sdk dependency. If we have one, label schema could be pickled.
BTW there is activity in ote on adding a demo script that can use tasks for inference. You can synchronize with it and probably unify the way of [serializing](https://github.com/openvinotoolkit/training_extensions/pull/737) of LabelSchemaEntity object. However, in SC environment a successor of LabelSchemaEntity is substituted everywhere, so downcasting back to LabelSchemaEntity could be required.

Reply:
I am aware of `LabelSchemaEntity` and it can have any structure. But as I see any inferencer using `List[LabelEntity]` and it is meaning that we already transpose complex label schema structure to linear structure and what about `LabelEntity` it have only one important field - name. So I think current implementation for openvino inferencer is cover label cases, because before saving labels to package or creating model from ModelAPI were called function `get_labels` for `task_environment.label_schema`. Maybe I don’t see the whole picture, in this case I can add serializing and according reading to demo or model wrappers.